File: c/fatorial.c
//f6 no vs code para rodar

#include <stdio.h>

// Função recursiva para calcular fatorial
int fatorialr(int n) {
    if (n == 0) {
        return 1;  // Caso base
    } else {
        return n * fatorialr(n - 1);  // Chamada recursiva
    }
}

int main() {
    int resultado = fatorialr(5);
    printf("%d\n", resultado);  // Saída: 120
    return 0;
}

File: python/fatorial.py

#fatorial: n! = n*(n-1)*(n-2)* ... *3*2*1 para n>0
#!0 = 1

'''
Sem utilizar recursão:

def fatorialr(n):
    resultado = 1
    for i in range(2, n+1): #i percorre os valores de 2 até n (incluso)
        resultado = resultado * i
    return resultado

print(fatorialr(0))
'''

def fatorialr(n):
    if n == 0:
        return 1
    else:
        return n*fatorialr(n-1)

print(fatorialr(5))



